product_category_company_favorite/tests/test_module.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (C) 2023-Today: GRAP (<http://www.grap.coop/>)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests import common
+
+
+class TestModule(common.TransactionCase):
+
+    def setUp(self):
+        super().setUp()
+        self.ProductCategory = self.env["product.category"]
+        self.ResCompany = self.env["res.company"]
+
+        self.main_company = self.env.ref("base.main_company")
+        self.demo_company = self.env.ref(
+            "product_category_company_favorite.demo_company"
+        )
+        self.category_A = self.env.ref("product.product_category_all")
+        self.category_A_2 = self.env.ref("product.product_category_1")
+        self.category_A_2_x = self.env.ref("product.product_category_5")
+
+    def _change_company(self, item, company=False):
+        if not company:
+            company = self.demo_company
+        return item.with_context(force_company=company.id)
+
+    def test_00_hook(self):
+        self.assertTrue(self.category_A.is_favorite)
+        self.assertTrue(self._change_company(self.category_A).is_favorite)
+
+    def test_10_write_value_company_dependant(self):
+        self.category_A.is_favorite = False
+        self.assertFalse(self.category_A.is_favorite)
+        self.assertTrue(self._change_company(self.category_A).is_favorite)
+
+    def test_11_write_value_recursive(self):
+        self.category_A.is_favorite = False
+        self.assertFalse(self.category_A.is_favorite)
+        self.assertFalse(self.category_A_2.is_favorite)
+        self.assertFalse(self.category_A_2_x.is_favorite)
+
+    def test_20_create_new_category(self):
+        root_categ_vals = {
+            "name": "New Root Category",
+        }
+        new_categ = self.ProductCategory.create(root_categ_vals)
+
+        self.assertFalse(new_categ.is_favorite)
+        self.assertTrue(self._change_company(new_categ).is_favorite)
```

Comment:
This makes no sense to me. Why is it not favourite for main_company, but it _is_ for demo_company?